asm/instruction.go

```diff
@@ -356,12 +398,34 @@ func (ins *Instruction) copyMetadata() *metadata {
 	return ins.metadata
 }
 
+// setMap sets the *ebpf.Map from which this instruction preforms a data.
+func (ins *Instruction) setMap(m FDer) {
+	if (ins.metadata != nil && ins.metadata.bpfMap == m) ||
+		(ins.metadata == nil && m == nil) {
+		return
+	}
+
+	ins.metadata = ins.metadata.copy()
+	ins.metadata.bpfMap = m
+}
+
+// FDer isn't actually used as a meaningful interface, rater it is used because we can't directly use types from the
```

Comment:
```suggestion
// FDer represents a resource tied to an underlying file descriptor.
// Used as a stand-in for e.g. ebpf.Map since that type cannot be
// imported here and FD() is the only method we rely on.
```